### PR TITLE
Refactor `unescapeJava()` usage in ballerina-lang

### LINKED ISSUE #42798: [Task]: Refactor `unescapeJava()` usage in ballerina-lang
### Description

The Ballerina Runtime currently uses a couple of different implementations of `unescapeJava()` method.
  1. org.apache.commons.text.StringEscapeUtils
  2. org.apache.commons.lang3.StringEscapeUtils
  3. io.ballerina.identifier.Utils

It would improve maintainability if there is only one implementation of the same functionality.

### Describe your task(s)

1. Remove the use of `org.apache.commons.text.StringEscapeUtils.unescapeJava()` and `org.apache.commons.lang3.StringEscapeUtils.unescapeJava()` from the codebase.
2. Replace the removed instances with `io.ballerina.identifier.Utils.unescapeJava()`

### Related area

-> Runtime

### Related issue(s) (optional)

#41530 

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by ravinperera00**
Fixes #42798 

## Approach
1. Removed the use of `org.apache.commons.text.StringEscapeUtils.unescapeJava()` and `org.apache.commons.lang3.StringEscapeUtils.unescapeJava()` from the codebase.
2. Replaced the removed instances with `io.ballerina.identifier.Utils.unescapeJava()`

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by gabilang — DISMISSED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/StateMachine.java [hunk 1]
```diff
@@ -18,10 +18,10 @@
 
 package io.ballerina.runtime.internal;
 
+import io.ballerina.identifier.Utils;
 import io.ballerina.runtime.api.creators.ErrorCreator;
 import io.ballerina.runtime.api.utils.StringUtils;
 import io.ballerina.runtime.api.values.BError;
-import org.apache.commons.lang3.StringEscapeUtils;
```
Comment (chiranSachintha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/StateMachine.java:24]: The `io.ballerina.identifier.Utils` class uses `StringEscapeUtils.unescapeJava`. We can replace that with `Utils.unescapeJava`.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -10580,7 +10579,7 @@ private BLangRecordLiteral rewriteMappingConstructor(BLangRecordLiteral mappingC
                 BLangSimpleVarRef varRefField = (BLangSimpleVarRef) field;
                 rewrittenFields.add(ASTBuilderUtil.createBLangRecordKeyValue(
                         rewriteExpr(createStringLiteral(pos,
-                                StringEscapeUtils.unescapeJava(varRefField.variableName.value))),
+                                Utils.unescapeJava(varRefField.variableName.value))),
```
Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:10582]: Better to refactor as follows
```
rewriteExpr(createStringLiteral(pos, Utils.unescapeJava(varRefField.variableName.value))),
```

Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:10582]: Please check other places as well to see if we can refactor them in this manner.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/StateMachine.java (hunk 1 above)
Comment (chiranSachintha) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/StateMachine.java:24]: Mistake from me. Within util class, we use `unescapeJava`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java (hunk 2 above)
Comment (ravinperera00) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:10582]: Got it. I'll refactor relevant places.

**Review by chiranSachintha — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

**Pull request merged**